services/autorust/codegen/src/identifier.rs

```diff
@@ -174,6 +174,8 @@ fn is_keyword(word: &str) -> bool {
             | "where"
             | "while"
             | "yield"
+        // required for Default::default support
```

Comment:
I don't understand why this change is required. What is not working?

Reply:
This removes the confusion between `::default()` and `.default`, which have wildly different meanings.  Consider `ResourceSkuCapacity::default()` verses `ResourceSkuCapacity.default`:

https://github.com/Azure/azure-sdk-for-rust/pull/602/files#diff-4169133b67bc7ca3e13582d656b1ece2bae7ce37d3be10b0c6d0076fdedc242f